swift/Multiplay/Multiplay.swift

```diff
@@ -3,39 +3,82 @@ import Foundation
 import SwiftyJSON
 
 func fetch(url:URL?) -> JSON? {
-    if url == nil {
-        return nil
-    }
+    guard let url = url else { return nil }
     let semaphore = DispatchSemaphore(value: 0)
     var json: JSON?
-    URLSession.shared.dataTask(with: url!) { data, res, err in
+    var timedOut = true
+    URLSession.shared.dataTask(with: url) { data, res, err in
         if let data = data {
             json = try? JSON(data: data)
+            print("Read ", data.count, " bytes\n")
+        }
+        else {
+            print("Failed to read url: ", url.absoluteString)
         }
+        timedOut = false
         semaphore.signal()
     }.resume()
     _ = semaphore.wait(timeout: DispatchTime.now() + DispatchTimeInterval.milliseconds(500))
+
+    if timedOut {
+        print("Request timed out for: ", url.absoluteString)
+    }
+    return json
+}
+
+func authenticate(url:URL?) -> JSON? {
+    guard let url = url else { return nil }
+    let request = URLRequest(url:url)
+    let config = URLSessionConfiguration.default
+    let userPassword = "StarLord89:69"
```

Comment:
Nice, perhaps a comment here on HTTP-Basic

Reply:
I imagine a tiny SwiftUI password dialog here would work.

Reply:
Oh dang, I left the testing code in. Yes, a dialog is required.